### ISSUE TITLE
Prevent phrases being duplicated

### ISSUE THREAD

**Issue by shaguniitb** [shaguniitb/mysite#26]

**Comment by jabbany** [comment]
This doesn't work...

We need to support having the same phrase in different categories for the following reasons:

- Categories are inherently not mutually exclusive. Forcing the user to pick only one category to place a phrase defeats the whole purpose of having freeform named categories as a way to organize phrases. (E.g. a phrase can be a swear word and also be an identity attack. The user should not be forced to place it in one or the other exclusively )
- Users may want to curate multiple shared lists/list templates that are subsets of each other intentionally. (E.g. "Swear words (Comprehensive)" + "Swear words (Basic)") Forcing a phrase to appear only on one list makes this impossible. 

Instead what we should do is:
- In general, come up with a precedence policy for what happens if a comment is matched by multiple filters (including duplicates)
- As a part of this, figure out precedence for the same phrase in multiple groups (what takes precedence)
- If a particular filter rule is superseded by another, show it as "inactive" in the UI, tell the user what supersedes it, but allow the user to keep it in a category.

Basically, for a "primary" instance, we might show:

    Category A:
    "word" (Also appears in 2 other categories) | Action: hold

And for a "superseded" instance, we show:

    Category B:
    "word" (Also appears in 2 other categories) | Action: inactive (Superseded by same filter in Category A)

**Comment by jabbany** [comment]
- In the UI when adding a rule check if it exists, show error if it exists.
- When cloning lists, skip any existing rules.

**Comment by jabbany** [comment]
- [ ] Come up with plan for which bits of code need updating (11/3)
- [ ] Implement backend check
    - [ ] Add rule
    - [ ] Clone list
- [ ] Implement frontend error messages
